baybe/surrogates/base.py

```diff
@@ -124,60 +134,80 @@ def _make_target_scaler() -> OutcomeTransform | None:
 
     def _make_input_scaler(self, searchspace: SearchSpace) -> ColumnTransformer:
         """Make the input scaler for transforming computational dataframes."""
-        from sklearn.compose import make_column_transformer
-
-        # Create the composite scaler from the parameter-wise scaler objects
-        transformers = [
-            (
-                "passthrough" if (s := self._make_parameter_scaler(p)) is None else s,
-                [c for c in p.comp_rep_columns if c in searchspace.comp_rep_columns],
-            )
-            for p in searchspace.parameters
-        ]
-        scaler = make_column_transformer(*transformers, verbose_feature_names_out=False)
-
-        scaler.fit(searchspace.comp_rep_bounds)
+        # Create a composite scaler from parameter-wise scaler objects
+        mapping: dict[tuple[int, ...], InputTransform] = {}
+        for p in searchspace.parameters:
+            idxs = searchspace.get_comp_rep_parameter_indices(p.name)
+            factory = self._make_parameter_scaler_factory(p)
+            if factory is None:
```

Comment:
COuldn't this be slightly optimized by having `if factory := self._make(....) is None: continue` as the very first line of this loop, even before calculating `idxs`?

Reply:
absolutely, thx! This happens when you write code that uses functionality that is not yet there (in this case `get_comp_rep_parameter_indices`) and then switch to implementing that functionality before finishing to optimize the control flow 🙃 Changed

Reply:
parenthesis missing, btw. Just noticed the problem and amended the commit